File: backend/src/main/java/vn/kurisu/mentormatch/service/NotificationService.java
package vn.kurisu.mentormatch.service;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import vn.kurisu.mentormatch.dto.response.ApiResponse;
import vn.kurisu.mentormatch.dto.response.NotificationResponse;
import vn.kurisu.mentormatch.entity.User;

public interface NotificationService {
    ApiResponse<Page<NotificationResponse>> getMyNotifications(Pageable pageable);
    ApiResponse<Void> markAsRead(Integer id);
    ApiResponse<Void> markAllAsRead();
    ApiResponse<Long> getUnreadCount();
    void sendNotification(User user, String title, String message, String type, Integer referenceId);
}

File: frontend/src/services/notificationService.js
import api from './api';

export const notificationService = {
  getMyNotifications: async (page = 0, size = 10) => {
    try {
      const response = await api.get(`/notifications?page=${page}&size=${size}`);
      return response;
    } catch (error) {
      console.error('Error fetching notifications:', error);
      throw error;
    }
  },

  getUnreadCount: async () => {
    try {
      const response = await api.get('/notifications/unread-count');
      return response;
    } catch (error) {
      console.error('Error fetching unread notification count:', error);
      throw error;
    }
  },

  markAsRead: async (id) => {
    try {
      const response = await api.put(`/notifications/${id}/read`);
      return response;
    } catch (error) {
      console.error('Error marking notification as read:', error);
      throw error;
    }
  },

  markAllAsRead: async () => {
    try {
      const response = await api.put('/notifications/read-all');
      return response;
    } catch (error) {
      console.error('Error marking all notifications as read:', error);
      throw error;
    }
  }
};
